log4j2-propagation/src/test/java/nl/talsmasoftware/context/log4j2/threadcontext/Log4j2ThreadContextManagerTest.java

```diff
@@ -89,29 +94,30 @@ public String call() {
 
     @Test
     void testProvider() {
-        assertSame(Log4j2ThreadContextManager.INSTANCE, Log4j2ThreadContextManager.provider());
+        Log4j2ThreadContextManager first = Log4j2ThreadContextManager.provider();
+        Log4j2ThreadContextManager second = Log4j2ThreadContextManager.provider();
+        assertThat(second, sameInstance(first));
+    }
 
-        // Verify that no-arg constructor, used by ServiceLoader, works
-        assertDoesNotThrow(new Executable() {
-            @SuppressWarnings("deprecation")
-            public void execute() {
-                new Log4j2ThreadContextManager();
-            }
-        });
+    @Test
+    @Deprecated
```

Comment:
Is it intended that this test method is `@Deprecated`? Since there is (probably) no intention to remove the constructor in the near future, would it make more sense to annotate this method with `@SuppressWarnings("deprecation")` instead?

Reply:
👍 You're right; that was laziness on my part 😉